tools/mason/MasonBuild.chpl

```diff
@@ -107,72 +107,64 @@ private proc checkChplVersion(lockFile : borrowed Toml) throws {
 proc buildProgram(release: bool, show: bool, force: bool, skipUpdate: bool,
                   ref cmdLineCompopts: list(string), tomlName="Mason.toml",
                   lockName="Mason.lock") throws {
+  const cwd = here.cwd();
+  const projectHome = getProjectHome(cwd, tomlName);
+  const toParse = open(projectHome + "/" + lockName, ioMode.r);
+  var lockFile = parseToml(toParse);
+  const projectName = lockFile["root"]!["name"]!.s;
 
-  try! {
+  // --fast
+  var binLoc = 'debug';
+  if release then
+    binLoc = 'release';
 
-    const cwd = here.cwd();
-    const projectHome = getProjectHome(cwd, tomlName);
-    const toParse = open(projectHome + "/" + lockName, ioMode.r);
-    var lockFile = parseToml(toParse);
-    const projectName = lockFile["root"]!["name"]!.s;
 
-    // --fast
-    var binLoc = 'debug';
-    if release then
-      binLoc = 'release';
+  // build on last modification
+  if projectModified(projectHome, projectName, binLoc) || force {
 
+    if isFile(projectHome + "/" + lockName) {
 
-    // build on last modification
-    if projectModified(projectHome, projectName, binLoc) || force {
+      // Make build files and check chapel version
+      makeTargetFiles(binLoc, projectHome);
+      checkChplVersion(lockFile);
 
-      if isFile(projectHome + "/" + lockName) {
-
-        // Make build files and check chapel version
-        makeTargetFiles(binLoc, projectHome);
-        checkChplVersion(lockFile);
-
-        if isDir(MASON_HOME) == false {
-          mkdir(MASON_HOME, parents=true);
-        }
+      if isDir(MASON_HOME) == false {
+        mkdir(MASON_HOME, parents=true);
+      }
 
-        // generate list of dependencies and get src code
-        var (sourceList, gitList) = genSourceList(lockFile);
+      // generate list of dependencies and get src code
+      var (sourceList, gitList) = genSourceList(lockFile);
 
-        if lockFile.pathExists('external') {
-          spackInstalled();
-        }
-        //
-        // TODO: Temporarily use `toArray` here because `list` does not yet
-        // support parallel iteration, which the `getSrcCode` method _must_
-        // have for good performance.
-        //
-        getSrcCode(sourceList, skipUpdate, show);
-
-        getGitCode(gitList, show);
-
-        // get compilation options including external dependencies
-        const compopts = getTomlCompopts(lockFile, cmdLineCompopts);
-        // Compile Program
-        if compileSrc(lockFile, binLoc, show, release, compopts, projectHome) {
-          writeln("Build Successful\n");
-        }
-        else {
-          throw new owned MasonError("Build Failed");
-        }
-        // Close memory
-        toParse.close();
+      if lockFile.pathExists('external') {
+        spackInstalled();
+      }
+      //
+      // TODO: Temporarily use `toArray` here because `list` does not yet
+      // support parallel iteration, which the `getSrcCode` method _must_
+      // have for good performance.
+      //
```

Comment:
You and I might have merge conflicts I am afraid. In my branch, I removed this comment, because I can't tell what its referring to. You can leave it as is, and we can race to see who merges first and who fixes the conflict. :D